clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -1613,14 +1615,18 @@ private ConsumerCoordinatorMetrics(Metrics metrics, String metricGrpPrefix) {
 
     private static class MetadataSnapshot {
         private final int version;
-        private final Map<String, Integer> partitionsPerTopic;
+        private final Map<String, List<RackInfo>> partitionsPerTopic;
```

Comment:
nit: I wonder if `PartitionInfo` would be a better name here. It would at least make clearer that we have one `PartitionInfo` object per partition. When I first read this, I was wondering why we have a list of RackInfo per topic.

Reply:
Since we already have another PartitionInfo, I renamed this one `PartitionRackInfo`.